File: files/400-499/427/CH-427.py
import pandas as pd

path = "400-499/427/CH-427 Custom Grouping.xlsx"
input = pd.read_excel(path, usecols="B:F", nrows=12, skiprows=2)
test = pd.read_excel(path, usecols="H:I", nrows=5, skiprows=2)

result = input.copy()
result["PRODUCTS"] = result[["PRODUCT 1", "PRODUCT 2", "PRODUCT 3"]].apply(
    lambda row: "-".join(sorted(row.dropna().astype(str))), axis=1
)
result = result.groupby("PRODUCTS").agg({"QUANTITY": "sum"}).reset_index()
result.columns = ["PRODUCTS", "TOTA; QUANTITY"]

# Result almost ideal. One set is not sorted in provided answer data.

File: files/400-499/427/CH-427.R
library(tidyverse)
library(readxl)

path <- "400-499/427/CH-427 Custom Grouping.xlsx"
input <- read_excel(path, range = "B3:F14")
test <- read_excel(path, range = "H3:I8")

result = input %>%
  mutate(
    PRODUCTS = pmap_chr(
      list(`PRODUCT 1`, `PRODUCT 2`, `PRODUCT 3`),
      ~ paste(sort(c(...)), collapse = "-")
    )
  ) %>%
  summarise(`TOTA; QUANTITY` = sum(QUANTITY), .by = PRODUCTS)

# Result almost ideal. One set is not sorted in provided answer data.
